ambry-router/src/main/java/com/github/ambry/router/GetBlobInfoOperation.java

```diff
@@ -353,9 +353,9 @@ private void handleBody(InputStream payload, MessageMetadata messageMetadata, Me
     ByteBuffer userMetadata = MessageFormatRecord.deserializeUserMetadata(payload);
     if (encryptionKey == null) {
       // if blob is not encrypted, move the state to Complete
-      operationResult =
-          new GetBlobResultInternal(new GetBlobResult(new BlobInfo(serverBlobProperties, userMetadata.array()), null),
-              null);
+      BlobInfo blobInfo = new BlobInfo(serverBlobProperties, userMetadata.array());
```

Comment:
It seems like in most places, `setLifeVersion` is called right after `new BlobInfo`. Instead of adding a new setter, one alternative could be making the BlobInfo.lifeVersion final and adding a new constructor that takes life version as an argument.